cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,292 @@
+"""
+cartography/intel/rapid7/util
+"""
+# pylint: disable=invalid-name,broad-except
+import json
+import logging
+import re
+from typing import Any
+from typing import List
+from typing import Tuple
+
+import pandas
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def extract_sub_from_resourceid(resourceid: str) -> str:
+    """
+    Get Azure subscription from Azure resource id
+    """
+    if isinstance(resourceid, str):
+        try:
+            match = re.search(
+                "/subscriptions/([0-9a-f-].*?)/resourceGroups/",
+                resourceid,
+                flags=re.IGNORECASE,
+            )
+            if match:
+                sub = match.group(1)
+                return sub
+        except AttributeError as exception:
```

Comment:
I don't think you need to catch this exception, since you already check if `match` is not `None`.